includes/payment-methods/class-wc-stripe-upe-payment-method.php

```diff
@@ -154,10 +154,20 @@ public function __call( $method, $arguments ) {
 
 		if ( in_array( $method, get_class_methods( $upe_gateway_instance ) ) ) {
 			return call_user_func_array( [ $upe_gateway_instance, $method ], $arguments );
-		} else {
+		}
+
+		// In development/staging environments, throw errors for undefined methods to help catch bugs.
+		// In production, fail gracefully to prevent third-party plugin compatibility issues.
+		$is_dev_environment = ( defined( 'WP_DEBUG' ) && WP_DEBUG )
+			|| in_array( wp_get_environment_type(), [ 'development', 'staging', 'local' ], true );
+
+		if ( $is_dev_environment ) {
 			$message = method_exists( $upe_gateway_instance, $method ) ? 'Call to private method ' : 'Call to undefined method ';
 			throw new \Error( $message . get_class( $this ) . '::' . $method );
 		}
+
+		WC_Stripe_Logger::warning( 'Call to undefined method ' . get_class( $this ) . '::' . $method );
```

Comment:
If logging is disabled for the Stripe plugin, we won't record this entry. I think we should upgrade it to an error so we always log the entry.
```suggestion
		WC_Stripe_Logger::error( 'Call to undefined method ' . get_class( $this ) . '::' . $method );
```